### PR TITLE
plugins/barbecue: migrate to mkNeovimPlugin

### PR DESCRIPTION
**Pull request by khaneliman**

### CODE REVIEW

File: plugins/bufferlines/barbecue.nix [hunk 1]
```diff
@@ -1,34 +1,72 @@
...
-in
-{
-  options.plugins.barbecue = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "barbecue-nvim";
+  inherit (lib.nixvim) defaultNullOpts;
 
-    package = helpers.mkPluginPackageOption "barbecue-nvim" pkgs.vimPlugins.barbecue-nvim;
-
-    attachNavic = helpers.defaultNullOpts.mkBool true ''
+  mkListStr = defaultNullOpts.mkNullable (lib.types.listOf lib.types.str);
```
Comment (MattSturgeon) [inline plugins/bufferlines/barbecue.nix:9]: 1. Is this abstraction really needed?
2. This doesn't support rawLua

I think we can just use `mkListOf types.str` throughout?

**Review by MattSturgeon — COMMENTED** [review]
No issues with the changes, but I don't think we should keep `mkListStr`. At least not as-is.

**Review by MattSturgeon — APPROVED** [review]
Can't see any issues :grinning:

**Comment by khaneliman** [comment]
@mergifyio queue

**Comment by MattSturgeon** [comment]
@mergifyio queue

**Pull request merged**